presto-native-execution/presto_cpp/main/PrestoExchangeSource.h

```diff
@@ -14,13 +14,50 @@
 #pragma once
 
 #include <folly/Uri.h>
+#include <folly/futures/Retrying.h>
 
 #include "presto_cpp/main/common/Configs.h"
 #include "presto_cpp/main/http/HttpClient.h"
 #include "velox/common/memory/Memory.h"
 #include "velox/exec/Exchange.h"
 
 namespace facebook::presto {
+
+class Backoff {
```

Comment:
Can we move this as a sub class of PrestoExchangeSource? And how about rename it to RetryState? And do you want to test the class separately?